{
  DefaultMuleConfiguration.verboseExceptions=true;
  MuleEvent testEvent=mock(MuleEvent.class);
  MuleMessage muleMessage=mock(MuleMessage.class);
  when(muleMessage.getMuleContext()).thenReturn(mockContext);
  Object payload=mock(Object.class);
  when(payload.toString()).then(new FailAnswer("toString() expected not to be called."));
  when(muleMessage.getPayload()).thenReturn(payload);
  when(transformationService.transform(muleMessage,DataType.STRING)).thenThrow(new TransformerException(CoreMessages.createStaticMessage("exception thrown")));
  when(testEvent.getMessage()).thenReturn(muleMessage);
  MessagingException e=new MessagingException(MessageFactory.createStaticMessage(message),testEvent);
  assertThat(e.getInfo().get(PAYLOAD_INFO_KEY),is(TransformerException.class.getName() + " while getting payload: exception thrown"));
}
